codeqlAlerts.js

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env node
+"use strict";
+
+require("dotenv").config();
+
+const _ = require("lodash");
+const moment = require("moment");
+const owner = "Sparkpost";
+const fileName = "alerts_" + Date.now() + ".json";
+const Promise = require("bluebird");
+const { Octokit } = require("@octokit/rest");
+const octokit = new Octokit({
+  auth: process.env.GITHUB_TOKEN,
+  // Set GitHub Auth Token in environment variable
+});
+
+const [, , ...args] = process.argv;
+
+//  .number, .created_at, .url, .html_url, .state, .dismissed_by.login, .dismissed_at, .dismissed_reason, .rule.id, .rule.severity, .rule.description, .tool.name, .most_recent_instance.classifications[]]
+async function getCodeAlerts(repos) {
+  return await Promise.map(repos, async ({ name, org }) => {
+    const sortedAlerts = {};
+    const summary = {};
+
+    return await octokit
+      .paginate(octokit.codeScanning.listAlertsForRepo, {
+        owner: org,
+        repo: name,
+      })
+      .then(async (alerts) => {
+        const filteredAlerts = await filterCodeAlerts(alerts);
+
+        for (var i = 0; i < filteredAlerts.length; i++) {
+          var rule = filteredAlerts[i].rule.description;
+          if (!sortedAlerts[rule]) {
+            sortedAlerts[rule] = { count: 1 };
+            sortedAlerts[rule]["createdAt"] = alerts[i].created_at;
+            sortedAlerts[rule]["severity"] = alerts[i].rule.severity;
+          } else {
+            sortedAlerts[rule]["count"] += 1;
+          }
+          summary["errorsCount"] =
+            filteredAlerts[i].rule.severity === "error"
+              ? summary["errorsCount"]
+              : (summary["errorsCount"] || 0) + 1;
+          summary["warningsCount"] =
+            filteredAlerts[i].rule.severity === "warning"
+              ? summary["warningsCount"]
+              : (summary["warningsCount"] || 0) + 1;
+        }
+        return sortedAlerts;
+      })
+      .then(async (sortedAlerts) => {
+        const blocks = [];
+        for (const alert in sortedAlerts) {
+          blocks.push(await buildBlocks("code", alert, sortedAlerts[alert]));
+        }
+        return { repo: name, summary, blocks };
+      });
+  }).catch((error) => {
+    console.error(
+      `Failed for ${org}/${name}\n${error.message}\n${error.documentation_url}`
+    );
+  });
+}
+
+async function filterCodeAlerts(alerts) {
+  return alerts.filter((alert) => {
+    const remediationDue = moment(alert.created_at).add(14, "days");
+    if (
+      alert.rule.severity !== "note" &&
+      alert.most_recent_instance.classifications !== "test" &&
+      remediationDue < moment()
+    ) {
+      return alert;
+    }
+  });
+}
+
+// .number, .html_url, .state, .secret_type, .secret, .resolution, .resolved_by, .resolved_at]
+async function getSecretAlerts(repos) {
```

Comment:
this one is tricky. while the Promise.map contains a bunch of async elements, this function does not wait for any of them.. in fact, since it returns a promise, it's expecting the calling function to handle its asynchronous waiting. as such, you shouldn't need async here.